src/components/modal/DeleteResourceModal.tsx

```diff
@@ -26,16 +26,19 @@ import { ComponentProps, createModalLauncher } from './createModalLauncher';
 type DeleteResourceModalProps = ComponentProps & {
   obj: K8sResourceCommon;
   model: K8sModelCommon;
+  displayName?: string;
   description?: React.ReactNode;
 };
 
 export const DeleteResourceModal: React.FC<DeleteResourceModalProps> = ({
   obj,
   model,
   onClose,
+  displayName,
   description,
 }) => {
   const [error, setError] = React.useState();
+  const resourceName = displayName || obj.metadata.name;
```

Comment:
Why not use it from resource object itself? 
```suggestion
  const resourceName = obj.spec.displayName ?? obj.metadata.name;
```

Reply:
I felt that passing name as a prop would be more generic, and we could use it for other resources too later on if needed.